src/v/config/configuration.h

```diff
@@ -184,7 +184,8 @@ struct configuration final : public config_store {
     property<std::vector<ss::sstring>> sasl_mechanisms;
     property<ss::sstring> sasl_kerberos_keytab;
     property<ss::sstring> sasl_kerberos_principal;
-    property<std::vector<ss::sstring>> sasl_kerberos_principal_mapping;
+    property<std::optional<std::vector<ss::sstring>>>
```

Comment:
If the default is `{{"DEFAULT"}}` is `std::optional<>` useful?

Reply:
I guess the user could set no rules.... just results in all kerberos authentications becoming anonymous.  So I don't know! :)  I can make it non-optional and then see who wants that feature later?

Reply:
I guess the point is whether setting it to `nullopt` is semantically different from setting it to an empty list.

Reply:
I guess we could leave it up to philosophers to decide this question